PR: https://github.com/odoo/odoo/pull/51658

From: 41bf013e04db0d03ca3648b70329df2c47087ef2
From: Odoo's Mergebot
Date: 2020-05-28 05:13:56

Structural Changes: 2
Total Changes: 54

[MERGE] base: make ir.property private

Only system users can access ir.property records.

This PR is part of the long-term task id 8203 that plans to remove all access to
 ir.* models

"Official" methods to interact with `ir.property`:
- `_set_default(name, model, value, company=False)`
- `_set_multi(name, model, values, default_value=None)`
- `_get(name, model, res_id=False)`
- `_get_multi(name, model, ids)`

Done in this PR:
- remove ACL on ir.property (only system users can access directly)
- make all getter/setters private with sudo
- use these methods everywhere instead of CRUD calls
- add sudo where needed

closes odoo/odoo#51658

Signed-off-by: Martin Trigaux (mat) <mat@odoo.com>

================================= pseudo patch: =================================

--- a/odoo/addons/base/data/base_data.sql
+++ b/odoo/addons/base/data/base_data.sql
@@ -84,13 +84,11 @@ CREATE TABLE ir_module_module_dependency (
 CREATE TABLE ir_model_data (
     id serial NOT NULL,
     create_uid integer,
-    create_date timestamp without time zone,
-    write_date timestamp without time zone,
+    create_date timestamp without time zone DEFAULT (now() at time zone 'UTC'),
+    write_date timestamp without time zone DEFAULT (now() at time zone 'UTC'),
     write_uid integer,
     noupdate boolean DEFAULT False,
     name varchar NOT NULL,
-    date_init timestamp without time zone,
-    date_update timestamp without time zone,
     module varchar NOT NULL,
     model varchar NOT NULL,
     res_id integer,

--- a/odoo/addons/base/models/ir_model.py
+++ b/odoo/addons/base/models/ir_model.py
@@ -1790,8 +1790,6 @@ class IrModelData(models.Model):
     module = fields.Char(default='', required=True)
     res_id = fields.Many2oneReference(string='Record ID', help="ID of the target record in the database", model_field='model')
     noupdate = fields.Boolean(string='Non Updatable', default=False)
-    date_update = fields.Datetime(string='Update Date', default=fields.Datetime.now)
-    date_init = fields.Datetime(string='Init Date', default=fields.Datetime.now)
     reference = fields.Char(string='Reference', compute='_compute_reference', readonly=True, store=False)
 
     _sql_constraints = [
@@ -1988,12 +1986,12 @@ class IrModelData(models.Model):
     # NOTE: this method is overriden in web_studio; if you need to make another
     #  override, make sure it is compatible with the one that is there.
     def _build_update_xmlids_query(self, sub_rows, update):
-        rowf = "(%s, %s, %s, %s, %s, now() at time zone 'UTC', now() at time zone 'UTC')"
+        rowf = "(%s, %s, %s, %s, %s)"
         return """
-            INSERT INTO ir_model_data (module, name, model, res_id, noupdate, date_init, date_update)
+            INSERT INTO ir_model_data (module, name, model, res_id, noupdate)
             VALUES {rows}
             ON CONFLICT (module, name)
-            DO UPDATE SET date_update=(now() at time zone 'UTC') {where}
+            DO UPDATE SET write_date=(now() at time zone 'UTC') {where}
         """.format(
             rows=", ".join([rowf] * len(sub_rows)),
             where="WHERE NOT ir_model_data.noupdate" if update else "",

--- a/odoo/addons/base/views/ir_model_views.xml
+++ b/odoo/addons/base/views/ir_model_views.xml
@@ -466,8 +466,8 @@
                                 <field name="module"/>
                                 <field name="name"/>
                                 <field name="noupdate"/>
-                                <field name="date_update" />
-                                <field name="date_init" />
+                                <field name="write_date" />
+                                <field name="create_date" />
                             </group>
                             <group>
                                 <field name="display_name"/>
